cmd/snap-confine/snap-confine.c

```diff
@@ -301,18 +302,25 @@ int main(int argc, char **argv)
 				}
 			}
 
-			// Associate each snap process with a dedicated snap freezer
-			// control group. This simplifies testing if any processes
-			// belonging to a given snap are still alive.
-			// See the documentation of the function for details.
+			// Associate each snap process with a dedicated snap freezer cgroup
+			// and snap pids cgroup. All snap processes belonging to one snap
+			// share the freezer cgroup. All snap processes belonging to one
+			// app or one hook share the pids cgroup.
+			//
+			// This simplifies testing if any processes belonging to a given
+			// snap are still alive as well as to properly account for each
+			// application and service.
 			if (getegid() != 0 && saved_gid == 0) {
-				// Temporarily raise egid so we can chown the freezer cgroup
-				// under LXD.
+				// Temporarily raise egid so we can chown the pids and freezer
+				// cgroups under LXD.
 				if (setegid(0) != 0) {
 					die("cannot set effective group id to root");
 				}
 			}
 			sc_cgroup_freezer_join(snap_instance, getpid());
+			if (sc_feature_enabled(SC_FEATURE_REFRESH_APP_AWARENESS)) {
+				sc_cgroup_pids_join(security_tag, getpid());
```

Comment:
Could we have a spread test at this point already? Just cat /sys...cgroup...pids ?

Reply:
Yes, I will add something quickly.

Reply:
NOTE: I didn't originally add a test because the feature was split and the security permissions related to this were separate. We can now do this because that patch has landed.